nativelib/src/main/scala/scala/scalanative/posix/sys/types.scala

```diff
@@ -0,0 +1,36 @@
+package scala.scalanative
+package posix.sys
+
+import scala.scalanative.native.{CInt, CStruct0, Ptr, extern}
+
+@extern
+object types {
+
+  type pthread_attr_t = Ptr[CStruct0]
+
+  type pthread_barrier_t = Ptr[CStruct0]
+
+  type pthread_barrierattr_t = Ptr[CStruct0]
+
+  type pthread_cond_t = Ptr[CStruct0]
+
+  type pthread_condattr_t = Ptr[CStruct0]
+
+  type pthread_key_t = Ptr[CStruct0]
+
+  type pthread_mutex_t = Ptr[CStruct0]
+
+  type pthread_mutexattr_t = Ptr[CStruct0]
+
+  type pthread_once_t = Ptr[CStruct0]
+
+  type pthread_rwlock_t = Ptr[CStruct0]
+
+  type pthread_rwlockattr_t = Ptr[CStruct0]
+
+  type pthread_spinlock_t = Ptr[CStruct0]
+
+  type pthread_t = Ptr[CStruct0]
```

Comment:
@densh all this types are opaque in the posix documentation, that's why I'd have used CStruct0 but most pthread functions take this direct types as arguments. What would be the best thing to do ?